openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIHeaderMapper.java

```diff
@@ -67,29 +70,48 @@
                 isOptional = values.get();
             }
         }
-        enableHeaderRequiredOption(headerParameter, node, stringSchema, isOptional);
-        completeHeaderParameter(parameters, headerName, headerParameter, stringSchema, headerParam.annotations(),
+        enableHeaderRequiredOption(headerParameter, node, headerTypeSchema, isOptional);
+        completeHeaderParameter(parameters, headerName, headerParameter, headerTypeSchema, headerParam.annotations(),
                 headerParam.typeName());
         return parameters;
     }
 
+    private String extractHeaderName(String headerParam) {
```

Comment:
It would be better to have the parameter node as the parameter of the util

Reply:
Here we are unable to send the parameter node since the parameter node has two types required, defaultable. Therefore I used to send the string. :)

Reply:
Sent a fix as discussed 🙂